Geometry/MTDNumberingBuilder/plugins/CmsMTDLevelBuilder.h

```diff
@@ -1,207 +1,49 @@
 #ifndef Geometry_MTDNumberingBuilder_CmsMTDLevelBuilder_H
-#define Geometry_MTDNumberingBuilder_CmsMTDLevelBuilder_H 
+#define Geometry_MTDNumberingBuilder_CmsMTDLevelBuilder_H
 
-#include "Geometry/MTDNumberingBuilder/plugins/CmsMTDAbstractConstruction.h"
+#include "FWCore/ParameterSet/interface/types.h"
 #include "Geometry/MTDNumberingBuilder/interface/CmsMTDStringToEnum.h"
 #include "Geometry/MTDNumberingBuilder/interface/GeometricTimingDet.h"
-#include "FWCore/ParameterSet/interface/types.h"
+#include "Geometry/MTDNumberingBuilder/plugins/CmsMTDAbstractConstruction.h"
 #include <string>
+
+class GeometricTimingDet;
+
+// it relies on the fact that the GeometricTimingDet::GDEnumType enumerators
+// used to identify the subdetectors in the upgrade geometries are equal to the
+// ones of the present detector + n*100
+bool subDetByType(const GeometricTimingDet* a, const GeometricTimingDet* b);
```

Comment:
These functions should probably be in a namespace

Reply:
But which namespace? The original functors before were public static members of the class. You think that's better?

Reply:
It's fine to have them as standalone functions, but they should be isolated in a namespace so they don't cause interference with other code. I would just make up a namespace like `mtd_helpers` or something. @lgray do you have a different preference?

Reply:
I'd rather have a shorter version: ```mtd```

Reply:
just mtd is probably good enough

Reply:
There is also the same problem for the TrackerLevelBuilder by the way. We should also not disregard the guidelines I think: https://cms-cpt-software.web.cern.ch/cms-cpt-software/General/CodingAndStyleRules.pdf

> 12. Do not use namespaces to group detector system classes e.g. tracker, calo etc.

I don't really know what to do. Personally I think it's also not the nicest solution to put these helper functions into `CmsMTDLevelBuilder`/`CmsTrackerLevelBuilder`, because they relate more to the `GeometricTimingDet`/`GeometricDet` class.

Reply:
actually, where else other than ```CmsMTDLevelBuilder``` are these functions used?
If nowhere, perhaps just move them to the class as static members?

Reply:
Ah sorry, I didn't read your comments before writing my own: if you all tell me it's ok with the namespace then I should consider that particular rule as outdated?

Reply:
And I agree with @slava77, that at this point the static members are probably the easiest solution.

Reply:
I think the code rule applies to detector system *classes*, not helper functions. The idea there is to make sure classes have descriptive names. For helper functions, namespaces are usually recommended.

Reply:
Ok I see, thank you! Anyway, I moved back to the static member version now, is that OK for you? This way, it's also closest to the original version with the functors.